dm/syncer/binlogstream/streamer_controller.go

```diff
@@ -277,41 +292,47 @@ func (c *StreamerController) resetReplicationSyncer(tctx *tcontext.Context, loca
 		c.streamProducer = &localBinlogReader{c.relay.NewReader(tctx.L(), &relay.BinlogReaderConfig{RelayDir: c.localBinlogDir, Timezone: c.timezone, Flavor: c.syncCfg.Flavor}), c.enableGTID}
 	}
 
-	c.upstream, err = newLocationedStream(c.streamProducer, location)
+	c.upstream, err = newLocationStream(c.streamProducer, location)
 	if err == nil {
 		c.streamModifier.reset(location)
+		c.locations.reset(location)
 		c.lastEventFromUpstream = nil
 	}
 	return err
 }
 
 // GetEvent returns binlog event from upstream binlog or streamModifier. It's not
 // concurrent safe.
-// When return events from streamModifier, we should maintain these properties:
+//
+// After GetEvent returns an event, GetCurStartLocation, GetCurEndLocation, GetTxnEndLocation
+// will return the corresponding locations of the event. The definition of 3 locations
+// can be found in the comment of locations struct in binlog_locations.go .
+//
+// When return events from streamModifier, 3 locations are maintained as below:
 //   - Inject
 //     if we inject events [DDL1, DDL2] at (start) position 900, where start position
 //     900 has Insert1 event whose LogPos (end position) is 1000, we should return
 //     to caller like
-//   - DDL1, start position 900 LogPos 900, suffix 1
-//   - DDL2, start position 900 LogPos 900, suffix 2
-//   - Insert1, start position 900 LogPos 1000, suffix 0
+//
+//     1. DDL1, start (900, suffix 0) end (900, suffix 1)
+//     2. DDL2, start (900, suffix 1) end (900, suffix 2)
+//     3. Insert1, start (900, suffix 2) end (1000, suffix 0)
```

Comment:
Maybe can clarify more about the meaning of (900, suffix 0) and the other similar expressions